presentation/handler/group_handler.go

```diff
@@ -52,6 +52,46 @@ func (h GroupHandler) Create(c *fiber.Ctx) error {
 	return core.Success(c, fiber.StatusCreated, SuccessMsgGroupCreate, group)
 }
 
+// Update updates a group with the given id.
+//
+//	@Summary	Update Group
+//	@Tags		Group
+//	@Accept		json
+//	@Produce	json
+//	@Param		id		path		string				true	"Group ID"
+//	@Param		request	body		dto.GroupInputDTO	true	"Request Body"
+//	@Success	200		{object}	dto.GeneralResponseDTO{data=dto.GroupDetailedOutputDTO}
+//
+//	@Router		/api/group/{id} [put]
+func (g GroupHandler) Update(c *fiber.Ctx) error {
+	// parse parameters
+	id := c.Params("id")
+	if id == "" {
+		return core.Error(c, fiber.StatusBadRequest, fmt.Sprintf(ErrMsgParameterRequired, "id"))
+	}
+
+	uid, err := uuid.Parse(id)
+	if err != nil {
+		return core.Error(c, fiber.StatusBadRequest, fmt.Sprintf(ErrMsgParseUUID, uid, err))
+	}
+
+	// parse request
+	var request GroupInputDTO
+	if err := c.BodyParser(&request); err != nil {
+		return core.Error(c, fiber.StatusBadRequest, fmt.Sprintf(ErrMsgGroupParse, err))
+	}
```

Comment:
Why do we use GroupInputDTO for the Update method. For Update Group we should be able to update all group fields not only the owner ID and the name.

Reply:
It's the same case as in `AddItem` and `ChangeItem`. There we send a `ItemInputDTO`, both for creating and updating an item. This is a common scenario that DTOs for creating and updating a resource are very similar (or equal). Using `GroupInputDTO` is not wrong here.

Group fields are:
- `ID` -> not contained in body
- `Name`  -> sent on POST and PUT
- `Owner`  -> sent on POST and PUT 
- `Bills`  -> not contained in body (add, update & remove bills handled via /bill endpoint)
- `Members`  -> missing on both POST and PUT

So the only field missing for an update is `Members` . We have 2 options: have two distinct DTOs (one for update, one for create) or one DTO with the missing Members. In general one DTO sounds better. A user can join a group only after accepting the invitation, so it's some extra logic we have to keep in mind. I don't want to deal with it here, I'll create a new issue for that.

Reply:
#108